crates/testing/src/view_generator.rs

```diff
@@ -0,0 +1,304 @@
+use std::marker::PhantomData;
+
+use hotshot_example_types::{
+    block_types::{TestBlockHeader, TestBlockPayload, TestTransaction},
+    node_types::{MemoryImpl, TestTypes},
+    state_types::TestInstanceState,
+};
+
+use crate::task_helpers::{
+    build_cert, build_da_certificate, build_vid_proposal, da_payload_commitment, key_pair_for_id,
+};
+use commit::Committable;
+
+use hotshot::types::{BLSPubKey, SignatureKey, SystemContextHandle};
+
+use hotshot_types::{
+    data::{Leaf, QuorumProposal, VidDisperse, ViewNumber},
+    message::Proposal,
+    simple_certificate::{DACertificate, QuorumCertificate, UpgradeCertificate},
+    simple_vote::{UpgradeProposalData, UpgradeVote},
+    traits::{
+        consensus_api::ConsensusApi,
+        node_implementation::{ConsensusTime, NodeType},
+    },
+};
+
+use hotshot_types::simple_vote::QuorumData;
+use hotshot_types::simple_vote::QuorumVote;
+
+#[derive(Clone)]
+pub struct TestView {
+    pub quorum_proposal: Proposal<TestTypes, QuorumProposal<TestTypes>>,
+    pub leaf: Leaf<TestTypes>,
+    pub view_number: ViewNumber,
+    pub quorum_membership: <TestTypes as NodeType>::Membership,
+    pub vid_proposal: (
+        Proposal<TestTypes, VidDisperse<TestTypes>>,
+        <TestTypes as NodeType>::SignatureKey,
+    ),
+    pub leader_public_key: <TestTypes as NodeType>::SignatureKey,
+    pub da_certificate: DACertificate<TestTypes>,
+    pub transactions: Vec<TestTransaction>,
+    upgrade_data: Option<UpgradeProposalData<TestTypes>>,
+}
+
+impl TestView {
+    pub fn genesis(quorum_membership: &<TestTypes as NodeType>::Membership) -> Self {
```

Comment:
This assumes DA = Quorum, I think for unit tests that's fine, but worth noting that if we ever want to test some scenario where the committee is smaller than the quorum we'll need both memberships here.